dashboard/app/jobs/concerns/active_job_metrics.rb

```diff
@@ -0,0 +1,104 @@
+require 'cdo/aws/metrics'
+require 'cdo/honeybadger'
+
+module ActiveJobMetrics
+  extend ActiveSupport::Concern
+
+  METRICS_NAMESPACE = 'code-dot-org/ActiveJob'.freeze
+
+  included do
+    # Parent class callbacks are called in addition to any callbacks defined in the job subclass
+    # Callback functions are executed in the following order:
+    #   before_enqueue and around_enqueue(before) in any order
+    #   Job is added to database or other queue store
+    #   after_enqueue and around_enqueue(after) in any order
+    #   before_perform and around_perform(before) in any order
+    #   job is performed
+    #   after_perform and around_perform(after) in any order
+    # When both the Parent and Child class define the same callback, they may be called in any order.
+    # Also note that jobs executed via `.perform_now` will not trigger the `enqueue` callbacks.
+    # https://guides.rubyonrails.org/v6.0/active_job_basics.html#available-callbacks
+    after_enqueue :report_job_count
+    before_perform :report_wait_time
+    after_perform :report_performance
+  end
+
+  protected def common_dimensions
+    [
+      {name: 'Environment', value: CDO.rack_env},
+      {name: 'JobName', value: self.class.name},
+      {name: 'JobId', value: job_id},
+      {name: 'QueueName', value: queue_name},
+    ]
+  end
```

Comment:
I'm not sure if `job_id` is a good idea here. This is more expensive and it makes looking at metrics in cloudwatch a bit more complicated because you have to create a custom Multi-Source query. You won't be able to discover problematic job_id's via metrics, you need to know the job ID first. Can we rely on Honeybadger for tracing failures instead of these metrics?

In terms of cost, we are billed per unique custom metric. We have 4 main metric types here: JobCount, WaitTime, ExecutionTime, TotalTime. This gets multiplied by each unique dimension - so 4 per JobName (2) per environment (4) per queue(2, but it's 1:1 with JobName, so counts as 1). That gets us up to 32. If we add a Job_id dimension, we increase to 32N, where N is the number of jobs we process per month, which is going to be a lot.

Reply:
Should I also remove `QueueName`?

Reply:
I think we'd be fine with just JobName